telesto/apps/segmentation.py

```diff
@@ -0,0 +1,158 @@
+import base64
+import io
+import os
+import socket
+import threading
+import time
+from importlib import import_module
+from pathlib import Path
+from typing import Type, Deque
+from uuid import uuid4
+import shutil
+import json
+from collections import deque
+
+from PIL import Image
+import falcon
+
+from telesto.models.segmentation import DummySegmentationModel, SegmentationModelBase
+from telesto.logger import logger
+
+
+def preprocess(doc: dict) -> Image:
+    image_bytes = base64.b64decode(doc["image"]["content"])
+    image = Image.open(io.BytesIO(image_bytes))
+    return image
+
+
+def image_to_base64(image: Image) -> bytes:
+    fp = io.BytesIO()
+    image.save(fp, format="PNG")
+    fp.seek(0)
+    return base64.b64encode(fp.read())
+
+
+def postprocess(image: Image) -> dict:
+    image_b64 = image_to_base64(image)
+    return {"mask": {"content": image_b64.decode()}}
+
+
+class SegmentationBase:
+    def on_get(self, req, resp):
+        doc = {"status": "ok", "host": socket.getfqdn(), "worker.pid": os.getpid()}
+        resp.body = json.dumps(doc, ensure_ascii=False)
+
+
+class ImageStorage:
+
+    def __init__(self, base_path: str = "/tmp/telesto/images"):
```

Comment:
I'd rather use [tempfile](https://docs.python.org/3/library/tempfile.html) here.

Reply:
Okay, but is it always going to run in a container or will someone try to run it on Windows?

Reply:
Good point, I always forget about people who code on Windows. I'll address it.